file/src/main/scala/akka/stream/alpakka/file/scaladsl/LogRotatorSink.scala

```diff
@@ -8,163 +8,197 @@ import java.nio.file.{OpenOption, Path, StandardOpenOption}
 
 import akka.Done
 import akka.stream.ActorAttributes.SupervisionStrategy
+import akka.stream.Supervision.Decider
 import akka.stream._
 import akka.stream.impl.fusing.MapAsync.{Holder, NotYetThere}
 import akka.stream.scaladsl.{FileIO, Sink, Source}
 import akka.stream.stage._
 import akka.util.ByteString
 
+import scala.collection.immutable
 import scala.concurrent.{ExecutionContext, Future, Promise}
 import scala.util.{Failure, Success}
 
 object LogRotatorSink {
+
+  /**
+   * Sink directing the incoming `ByteString`s to new files whenever `generatorFunction` returns a new value.
+   *
+   * @param functionGeneratorFunction creates a function that triggers rotation by returning a value
+   * @param fileOpenOptions file options for file creation
+   */
   def apply(
       functionGeneratorFunction: () => ByteString => Option[Path],
```

Comment:
I see that this has not changed in this PR, but do you see a need for the additional wrapping with a function? Could this be `ByteString => Option[Path]` instead?

Reply:
Scratch that. I see that it is used to capture some state in the `fileLengthFunctionFactory` example below.